test/datatable.jl

```diff
@@ -234,140 +232,142 @@ module TestDataTable
 
     @test append!(DataTable(A = 1:2, B = 1:2), DataTable(A = 3:4, B = 3:4)) == DataTable(A=1:4, B = 1:4)
-    dt = DataTable(A = NullableArray(1:3), B = NullableArray(4:6))
-    @test all(c -> isa(c, NullableArray), categorical!(deepcopy(dt)).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1,2]).columns)
-    @test all(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A,:B]).columns)
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [:A]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), :A).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), [1]).columns) == [1]
-    @test find(c -> isa(c, NullableCategoricalArray), categorical!(deepcopy(dt), 1).columns) == [1]
+    dt = DataTable(A = 1:3, B = 4:6)
+    @test all(c -> isa(c, Vector), categorical!(deepcopy(dt)).columns)
+    @test all(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [1,2]).columns)
+    @test all(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [:A,:B]).columns)
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [:A]).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), :A).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), [1]).columns) == [1]
+    @test find(c -> isa(c, CategoricalArray), categorical!(deepcopy(dt), 1).columns) == [1]
 
     @testset "unstack nullable promotion" begin
         dt = DataTable(Any[repeat(1:2, inner=4), repeat('a':'d', outer=2), collect(1:8)],
                        [:id, :variable, :value])
         udt = unstack(dt)
-        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
-        @test udt == DataTable(Any[Nullable[1, 2], Nullable[1, 5], Nullable[2, 6],
-                                   Nullable[3, 7], Nullable[4, 8]], [:id, :a, :b, :c, :d])
-        @test all(typeof.(udt.columns) .== NullableVector{Int})
+        #FIXME
+        # @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
+        @test udt == DataTable(Any[[1, 2], [1, 5], [2, 6],
+                                   [3, 7], [4, 8]], [:id, :a, :b, :c, :d])
+        @test all(typeof.(udt.columns) .== Vector{?Int})
         dt = DataTable(Any[categorical(repeat(1:2, inner=4)),
                            categorical(repeat('a':'d', outer=2)), categorical(1:8)],
                        [:id, :variable, :value])
         udt = unstack(dt)
-        @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
-        @test udt == DataTable(Any[Nullable[1, 2], Nullable[1, 5], Nullable[2, 6],
-                                   Nullable[3, 7], Nullable[4, 8]], [:id, :a, :b, :c, :d])
+        # @test udt == unstack(dt, :variable, :value) == unstack(dt, :id, :variable, :value)
```

Comment:
OK, the trace shows that there's a problem in DataTables: one should never try to create an `Array` holding `CategoricalValue` objects. I'm not sure how we missed it (I guess it worked, though it was kind of absurd), but we definitely need a special handling in `nullable!`. The simplest way do to that is probably to make it call `dt[col] = copy!(similar_nullable(dt[col]), dt[col])`.